From b872b2f9f20b0677b50293dda7e683712a140abe Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Sun, 17 Apr 2022 09:38:17 +0000
Subject: [PATCH] libstdc++: fix DCHECK_EQ/NE in
 content::FirstPartySetsDatabase

There is no CheckOpValueStr() for std::unique_ptr. Use get() in
DCHECK_EQ to retrieve the pointer value. Change DCHECK_NE to
DCHECK.
---
 .../database/first_party_sets_database.cc                 | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/content/browser/first_party_sets/database/first_party_sets_database.cc b/content/browser/first_party_sets/database/first_party_sets_database.cc
index 54e36e4..18a9034 100644
--- a/content/browser/first_party_sets/database/first_party_sets_database.cc
+++ b/content/browser/first_party_sets/database/first_party_sets_database.cc
@@ -173,7 +173,7 @@ bool FirstPartySetsDatabase::LazyInit() {
   if (db_status_ != InitStatus::kUnattempted)
     return db_status_ == InitStatus::kSuccess;
 
-  DCHECK_EQ(db_, nullptr);
+  DCHECK_EQ(db_.get(), nullptr);
   db_ = std::make_unique<sql::Database>(sql::DatabaseOptions{
       .exclusive_locking = true, .page_size = 4096, .cache_size = 32});
   db_->set_histogram_tag("FirstPartySets");
@@ -196,7 +196,7 @@ bool FirstPartySetsDatabase::LazyInit() {
 }
 
 bool FirstPartySetsDatabase::OpenDatabase() {
-  DCHECK_NE(db_, nullptr);
+  DCHECK(db_);
   if (db_->is_open() || db_->Open(db_path_)) {
     db_->Preload();
     return true;
@@ -206,7 +206,7 @@ bool FirstPartySetsDatabase::OpenDatabase() {
 
 void FirstPartySetsDatabase::DatabaseErrorCallback(int extended_error,
                                                    sql::Statement* stmt) {
-  DCHECK_NE(db_, nullptr);
+  DCHECK(db_);
   // Attempt to recover a corrupt database.
   if (sql::Recovery::ShouldRecover(extended_error)) {
     // Prevent reentrant calls.
@@ -329,4 +329,4 @@ bool FirstPartySetsDatabase::Destroy() {
   return base::DeleteFile(db_path_);
 }
 
-}  // namespace content
\ No newline at end of file
+}  // namespace content
-- 
2.35.1

